### PR TITLE
Ban problematic OpenRouter upstream providers

### PR DESCRIPTION
**Pull request by 40min**
## Summary
- add OPENROUTER_DISALLOWED_PROVIDERS setting to configure a provider blacklist
- include OpenRouter provider ignore routing in both agent and service LLM builders
- add tests for config parsing and OpenRouter payload wiring
- document the new env var in .env.example and .env.test

## Validation
- make backend-test (918 passed)
- uv run --extra dev pytest tests/agents/test_llm.py tests/core/test_factory.py tests/test_config.py -q (41 passed)

## Context
This addresses the OpenRouter provider glitch noted in Dart task: https://app.dartai.com/t/4ZP47a8Gkry6-bug-provider-glitch-on-openrou

### CODE REVIEW

File: src/runestone/config.py [hunk 1]
```diff
@@ -204,6 +205,12 @@ def resolve_ocr_llm_model(self) -> str:
             return self.ocr_llm_model_name
         return self.resolve_service_llm_model(provider=self.resolve_ocr_llm_provider())
 
+    def resolve_openrouter_disallowed_providers(self) -> list[str]:
+        """Return normalized OpenRouter providers to avoid routing through."""
+        if not self.openrouter_disallowed_providers:
+            return []
+        return [provider.strip() for provider in self.openrouter_disallowed_providers.split(",") if provider.strip()]
```
[the remark this answers is not in the document]
Comment (40min) [reply src/runestone/config.py]: Implemented. `resolve_openrouter_disallowed_providers()` now deduplicates slugs while preserving order using `dict.fromkeys(...)`. I also updated the config test to cover duplicate input and reran targeted checks: `uv run --extra dev pytest tests/test_config.py tests/agents/test_llm.py tests/core/test_factory.py -q` (41 passed).
